### PR TITLE
Update prettier and eslint for optional chaining support

### PR DESCRIPTION
**Pull request by amcasey**
Follow up to #10025.  This should make it possible to use features like `?.`.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @rafeca

**Comment by rafeca** [comment]
Awesome! Now that both the TypeScript and Node.js PRs have been upgraded if you rebase this PR hopefully CI will pass 🤞

File: package.json [hunk 1]
```diff
@@ -172,4 +172,4 @@
     "electron-packager": "^14.2.1",
     "electron-winstaller": "4.0.0"
   }
-}
+}
```
Comment (rafeca) [inline package.json]: Let's keep the end line here to not have it changed back in another PR

```suggestion
}

```

File: .prettierrc.yml [hunk 2]
```diff
@@ -2,6 +2,8 @@ singleQuote: true
 trailingComma: es5
 semi: false
 proseWrap: always
+endOfLine: auto
```
Comment (rafeca) [inline .prettierrc.yml:5]: Does keeping the default value for `endOfLine` produce lots of changes in the PR? If not, I prefer to have the default value (`lf`) to not have mixed end of line flavors.

```suggestion
```

File: .eslintrc.yml [hunk 3]
```diff
@@ -24,10 +24,18 @@ rules:
...
+  '@typescript-eslint/naming-convention':
     - error
-    - always
-  '@typescript-eslint/no-angle-bracket-type-assertion': error
+    - selector: interface
+      format:
+        - PascalCase
+      custom:
+        regex: '^I[A-Z]'
+        match: true
```
Comment (rafeca) [inline .eslintrc.yml:34]: Nice! Much more explicit now!

File: package.json (hunk 1 above)
Comment (amcasey) [reply package.json]: I have no preference - I leave it in whatever state it's in after `lint:fix` runs.

File: .prettierrc.yml (hunk 2 above)
Comment (amcasey) [reply .prettierrc.yml:5]: Yes, the diff was huge, so I set it back to the old value.

File: .eslintrc.yml (hunk 3 above)
Comment (amcasey) [reply .eslintrc.yml:34]: Stolen from the typescript-eslint docs. :wink:

File: .prettierrc.yml (hunk 2 above)
Comment (amcasey) [reply .prettierrc.yml:5]: If I'm reading your comment correctly, it sounds like you're on board with saving this change for a future PR.

File: package.json (hunk 1 above)
Comment (amcasey) [reply package.json]: And it looks like `lint:fix` took care of it in the second commit. :smile:

**Comment by amcasey** [comment]
@rafeca Maintainer edits are enabled, so please feel free to make whatever changes you see fit.  From my perspective, things look good.

**Comment by rafeca** [comment]
Thanks for the response! I think we can merge this one then 😃

File: package.json (hunk 1 above)
Comment (rafeca) [reply package.json]: Ah cool, nevermind then 😄

**Pull request merged**